transport-classes-kqueue/src/main/java/io/netty/channel/kqueue/KQueueEventLoop.java

```diff
@@ -167,8 +171,9 @@ private int kqueueWait(boolean oldWakeup) throws IOException {
         }
 
         long totalDelay = delayNanos(System.nanoTime());
-        int delaySeconds = (int) min(totalDelay / 1000000000L, Integer.MAX_VALUE);
-        return kqueueWait(delaySeconds, (int) min(totalDelay - delaySeconds * 1000000000L, Integer.MAX_VALUE));
+        int delaySeconds = (int) min(totalDelay / 1000000000L, KQUEUE_MAX_TIMEOUT_SECONDS);
+        int delayNanos = (int) (totalDelay % 1000000000L);
```

Comment:
Don't we also need to subtract the delaySeconds ?

Reply:
@ikhoon ping

Reply:
Agree, it seems a typo here

Reply:
Wrote a unit test and it passed.
```java
@Test
void testNanos() {
    final long nanoTime = System.nanoTime();
    final long seconds = nanoTime / 1_000_000_000L;
    final long nanos1 = nanoTime % 1_000_000_000L;
    final long nanos2 = nanoTime - seconds * 1_000_000_000L;
    assertThat(nanos2).isEqualTo(nanos1);
}
```

Reply:
This is fine because `KQUEUE_MAX_TIMEOUT_SECONDS` is 24 hours minus 1 second, and the nanosecond component cannot exceed one second. Keeping the total under 24 hours.